src/Types.jl

```diff
@@ -34,9 +34,9 @@ abstract type AbstractConstraint end
 
 abstract type AbstractCache end
 
+abstract type AbstractStoppingCondition end
 
 @enum ProblemSense Min Max
-@enum OptimizationStatus Unoptimized PrecisionLimit IterationLimit
```

Comment:
We should keep the status enum around and return these as well. It is far easier to test against enum values programmatically than test against the strings (you have to know what the strings say). The strings should only be for human consumption in the form of logging messages.

Reply:
Unless we want to treat precision limit and iteration limits as special cases (or limit this to something like "unoptimized, infeasible solutions, feasible solutions" etc)  then is an enum the best solution? They cannot be extended with new values interactively. Could directly using the stopping condition types be suitable instead?

Reply:
I think we should provide some general status codes here that are applicable for multiple conditions in this enum, but then have an enum value for `OTHER_TERMINATION_CONDITION` that can be used for those custom ones that don't quite fit anywhere else. That is also where the status string comes into play.

Reply:
I've added a possible implementation of this in 541a321, the enum behaves as before, but will display that the stopping condition is unknown if it terminates for any reason other than reaching the iteration/precision limit. I think it's safe to treat these two as slightly special cases as they should generally be present (but not having them won't break anything).